src/Controls/src/BindingSourceGen/BindingSourceGenerator.cs

```diff
@@ -115,12 +115,18 @@ static BindingDiagnosticsWrapper GetBindingForGeneration(GeneratorSyntaxContext
 		{
 			diagnostics.Add(Diagnostic.Create(
 				DiagnosticsDescriptors.UnableToResolvePath, lambda.Body.GetLocation(), lambda.Body.ToString()));
+			return new BindingDiagnosticsWrapper(null, diagnostics.ToArray());
 		}
 
+		// Sometimes analysing just the return type of the lambda is not enough. TODO: Refactor
+		var propertyType = CreateTypeNameFromITypeSymbol(lambdaSymbol.ReturnType, enabledNullable);
+		var lastMember = parts.Last() is Cast cast ? cast.Part : parts.Last();
```

Comment:
I don't like this, this will definitely break at some point. I think we should add `bool IsConditional` to `IPathPart` instead and implement the logic there.

Reply:
Yes, I think we need to consider the whole path, like for example in `f.A?.B.C.D` the property type should be nullable even though the last part itself isn't conditional.

Originally I was thinking more about the order of nesting - would there be a case in which `ConditionalAcecss` wouldn't be directly under `Cast`? In that case, I think we need https://github.com/simonrozsival/maui/pull/13/files#diff-629699fb7ba110fe45a77f9d06ea6a4ddb5f8d506a3017fc6123c3730b578f84L174-R174.

> This will not work if we add support for `??`

I don't think we'll want to support `??` anytime soon. If we add it, we'll revisit it.

Reply:
As we decided on what `CondtionalAccess` should mean, we will encounter one problem with this part.

Let us consider 

```
        using Microsoft.Maui.Controls;
        var label = new Label();
        label.SetBinding(Label.RotationProperty, static (Foo f) => f.Value);

        class Foo
        {
            public string? Value { get; set; }
        }
```

The problem here is that as discussed we are not able to determine the `NullableAnnotations` on lambda return type. It will be `NullableAnnotations.None` unless we add `string?` generic parameter.

The current workaround for this was that we inspected Path. However as we decided that `f.Value` is not a conditional access. The path will not contain an instance of `ConditionalAccess`. 

So I guess that either we can add a new property to the IPathPart `CanBeNull` which basically informs us whether the current member can be null. Or we can manually inspect the type of the last element of the path. (If it is not last element it should work without workaround)

Reply:
This is out of scope of this PR. Commented out the functionality and flagged the test as SKIP.